modules/components/src/AdvancedFacetView/index.js

```diff
@@ -1,26 +1,48 @@
 import React from 'react';
-import { keys } from 'lodash';
+import { keys, debounce, isEqual } from 'lodash';
 import { Subject } from 'rxjs';
 import { truncate } from 'lodash';
+import { css } from 'emotion';
 import NestedTreeView from '../NestedTreeView';
 import SQONView, { Bubble, Field, Value } from '../SQONView';
 import './AdvancedFacetView.css';
 import FacetView from './FacetView';
 import { replaceSQON, toggleSQON } from '../SQONView/utils';
-import SearchBox from './SearchBox';
+import Input from '../Input';
+import Component from 'react-component-component';
 import {
   filterOutNonValue,
   injectExtensionToElasticMapping,
   orderDisplayTreeData,
+  filterDisplayTreeDataBySearchTerm,
 } from './utils.js';
+import TextInput from '../Input';
+import FaFilter from 'react-icons/lib/fa/filter';
+import Spinner from 'react-spinkit';
+import FaTimesCircleO from 'react-icons/lib/fa/times-circle-o';
+
+const spinner = (
+  <Spinner
+    fadeIn="none"
+    name="circle"
+    color="#a9adc0"
+    style={{
+      width: 30,
+      height: 30,
+    }}
+  />
+);
```

Comment:
Is this spinner applicable to other parts of arranger? If so it might be worth giving it its own file so it's obvious for re-use.